src/jdk.jfr/share/classes/jdk/jfr/events/SerializationMisdeclarationEvent.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.jfr.events;
+
+import jdk.jfr.Category;
+import jdk.jfr.Description;
+import jdk.jfr.Label;
+import jdk.jfr.Name;
+import jdk.jfr.internal.MirrorEvent;
+import jdk.jfr.internal.RemoveFields;
+import jdk.jfr.internal.Type;
+
+@Name(Type.EVENT_NAME_PREFIX + "SerializationMisdeclaration")
+@Label("Serialization Deserialization")
+@Category({"Java Development Kit", "Serialization"})
+@Description("Methods and fields misdeclarations")
+@MirrorEvent(className = "jdk.internal.event.SerializationMisdeclarationEvent")
+@RemoveFields({"duration", "stackTrace", "thread"})
+public final class SerializationMisdeclarationEvent extends AbstractJDKEvent {
+
+    @Label("Class")
+    public Class<?> cls;
+
+    @Label("Kind")
+    public int kind;
```

Comment:
What is the use case for error codes? Are they public or an implementation detail?

Reply:
The intent is that they are stable and for programmatic usage, whereas the message is more for human consumption. The codes are used in the test, for example, and are declared as public static in the event classes.

Alternatively, one could parse the message, but that's less robust in face of changes, I think.

Reply:
Users (not OpenJDK developers) don't know what the error code means. I think it's better to not have them. This is how other events work. If you want to guard against changes, I would export the package to the test.

Reply:
What about fixed `String`s rather than `int`s for the kind of error?
Something like `"SUID_INEFFECTIVE_ON_ENUM"`, and so on?
It would be nice to be able to use enums, but AFAIK that's not supported in JFR.

Reply:
You could define them with an Enum but use the ordinal as the value for JFR.

Reply:
Same remark here about finality. public statics should be final.

Reply:
Working on that.